File: problems/102/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (not root) {
            return {};
        }
        
        vector<vector<int>> ans;
        queue<TreeNode*> q;
        
        q.push(root);
        
        while (not q.empty()) {
            ans.push_back({});
            
            for (int i = q.size(); i > 0; i--) {
                ans.back().push_back(q.front()->val);
                
                if (q.front()->left) {
                    q.push(q.front()->left);
                }
                
                if (q.front()->right) {
                    q.push(q.front()->right);
                }
                
                q.pop();
            }
        }
        
        return ans;
    }
};

File: problems/102/solution.go
package main
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
    if root == nil {
        return [][]int{}
    }
    
    ans, q := [][]int{}, []*TreeNode{root}
    
    for len(q) > 0 {
        curLevel, newQ := []int{}, []*TreeNode{}
        
        for _, node := range q {
            curLevel = append(curLevel, node.Val)
            
            if node.Left != nil {
                newQ = append(newQ, node.Left)
            }
            
            if node.Right != nil {
                newQ = append(newQ, node.Right)
            }
        }
        
        ans = append(ans, curLevel)
        q = newQ
    }
    
    return ans
}